test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -2391,6 +2391,31 @@ def forward(self, x):
         x = torch.randn(2, 3, 5, 5)
         self.run_test(LogDet(), x)
 
+    def test_dim(self):
+        class DimModel(torch.jit.ScriptModule):
+            @torch.jit.script_method
+            def forward(self, input):
+                out = input * 2
+                out *= out.dim()
+                return out
+        empty_input = torch.randn(0, requires_grad=True)
+        multi_dim_input = torch.randn(1, 2, 3, requires_grad=True)
+        self.run_test(DimModel(), empty_input)
+        self.run_test(DimModel(), multi_dim_input)
```

Comment:
could you specify "test_with_inputs" with an input of different shape to one of the self.run_test()?

This will run a test on the exported model with the additional input, to verify that the same exported model will run with inputs of different sizes (it would be a good test in this case, since getting the output size info is the main purpose of this op).

Reply:
Since rank must be constant for input there is no way to input a tensor of a different rank to test with thus we cannot truly test that way. Discussed offline with Lara, will resolve this.